pkg/controller/deployment/deployment_controller.go

```diff
@@ -440,6 +443,18 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	return fmt.Errorf("unexpected deployment strategy type: %s", d.Spec.Strategy.Type)
 }
 
+// Updates the status of a paused deployment
+func (dc *DeploymentController) syncPausedDeploymentStatus(deployment *extensions.Deployment) error {
+	newRS, oldRSs, err := dc.getAllReplicaSets(*deployment, false)
+	if err != nil {
+		return err
+	}
+	allRSs := append(controller.FilterActiveReplicaSets(oldRSs), newRS)
+
```

Comment:
Add the following stanza here so that cleanup policy for paused deployments (https://github.com/kubernetes/kubernetes/issues/20966) won't be blocked on https://github.com/kubernetes/kubernetes/pull/20273
```go
if deployment.Spec.RevisionHistoryLimit != nil {
 		// Cleanup old replica sets
 		dc.cleanupOldReplicaSets(oldRSs, deployment)
 	}
```